lib/ts/recipe/multifactorauth/recipeImplementation.ts

```diff
@@ -396,28 +405,23 @@ export default function getRecipeInterface(
             };
         },
 
-        createOrUpdateSessionForMultifactorAuthAfterSignIn: async function (
+        createOrUpdateSessionForMultifactorAuthAfterFactorCompletion: async function (
             this: RecipeInterface,
-            {
-                req,
-                res,
-                tenantId,
-                factorIdInProgress,
-                justSignedInUser,
-                justSignedInUserCreated,
-                justSignedInRecipeUserId,
-                userContext,
-            }
+            { req, res, tenantId, factorIdInProgress, justCompletedFactorUserInfo, userContext }
         ) {
-            const session = await Session.getSession(req, res, { sessionRequired: false });
+            let session = await Session.getSession(req, res, { sessionRequired: false });
             if (
                 session === undefined // no session exists, so we can create a new one
             ) {
+                if (justCompletedFactorUserInfo === undefined) {
+                    throw new Error("should never come here"); // We wouldn't create new session from a recipe like TOTP
+                }
+
                 const session = await Session.createNewSession(
```

Comment:
This shadows the session var from the scope above. It doesn't cause in issue, but it's a bit confusing.

Reply:
Renamed for clarity